python_modules/dagster/dagster/_core/definitions/partition_mapping.py

```diff
@@ -551,6 +555,338 @@ def get_downstream_partitions_for_partitions(
             )
 
 
+class DimensionMapping(
+    NamedTuple(
+        "_DimensionMapping",
+        [
+            ("upstream_dimension_name", str),
+            ("downstream_dimension_name", str),
+            ("partition_mapping", PartitionMapping),
+        ],
+    )
+):
+    """
+    Defines an explicit correspondence between one dimension of a MultiPartitionsDefinition
+    and one dimension of another MultiPartitionsDefinition.
+
+    Args:
+        upstream_dimension_name (str): The name of the dimension in the upstream MultiPartitionsDefinition.
+        downstream_dimension_name (str): The name of the dimension in the downstream MultiPartitionsDefinition.
+        partition_mapping (PartitionMapping): The partition mapping object used to calculate
+            the downstream dimension partitions from the upstream dimension partitions and vice versa.
+    """
+
+    def __new__(
+        cls,
+        upstream_dimension_name: str,
+        downstream_dimension_name: str,
+        partition_mapping: PartitionMapping,
+    ):
+        return super(DimensionMapping, cls).__new__(
+            cls,
+            upstream_dimension_name=check.str_param(
+                upstream_dimension_name, "upstream_dimension_name"
+            ),
+            downstream_dimension_name=check.str_param(
+                downstream_dimension_name, "downstream_dimension_name"
+            ),
+            partition_mapping=check.inst_param(
+                partition_mapping, "partition_mapping", PartitionMapping
+            ),
+        )
+
+
+@experimental
+@whitelist_for_serdes
+class MultiPartitionsMapping(
```

Comment:
Since this is a subclass of `PartitionMapping`, I would expect it to end in `PartitionMapping` like the other subclasses (instead of `PartitionsMapping`). Let me know if you think otherwise.

Reply:
I'm fine with naming it `MultiPartitionMapping`